app/konnect/gateway-manager/plugins/update-custom-plugin.md

```diff
@@ -0,0 +1,142 @@
+---
+title: Editing or deleting a custom plugin's schema
+content_type: how-to
+---
+
+The workflow for updating an in-use custom plugin depends on whether you need to update the schema:
+
+* **No change to plugin schema:** Editing a custom plugin's logic **without** any change in 
+schema doesn't cause the control plane to go out of sync with the data planes. 
+
+  In this situation, you only need to make sure that each data plane node has the correct logic. 
+  The schema on the control plane doesn't need to be updated.
+
+* **Changes to plugin schema:** If the schema needs to change, you must update both the schema in 
+{{site.konnect_short_name}} and the plugin code itself on each data plane node.
+
+There is no versioning for custom plugins. 
+If you need to version a schema (that is, maintain two or more similar copies of a custom plugin), upload it as a new custom plugin with a unique name. 
+For example, if your original plugin is named `delay`, you would have to name the copy 
+something else, such as `delay-v2`.
+
+## Updating a custom plugin
+
+### How the {{site.konnect_short_name}} platform reads configuration
+
+When a schema is updated in {{site.konnect_short_name}}, the {{site.konnect_short_name}} 
+platform doesn't trigger payload reconciliation automatically.
+This means that if you don't make any configuration changes, such as adding, 
+modifying, or deleting a {{site.base_gateway}} entity, the data plane nodes won't receive a
+payload update.
+
+When pushing changes to the control plane, the payload reconciliation only affects 
+data plane nodes if an instance of the plugin that uses the updated schema has its configuration 
+changed.
+
+Since plugin configurations are stored as JSON blobs, a schema change alone doesn't impact the 
+plugin configuration. However, if the plugin itself is updated, the new schema affects how 
+the new plugin configuration is represented.
+
+### Custom plugin update path
+
+When you need to make plugin changes, we recommend updating the schema in 
+{{site.konnect_short_name}} first, and then on the data plane nodes:
+
+1. Start a migration window.
```

Comment:
This is outside of Kong's control. It's a process step - ie, if you're a Kong admin, to be safe, you would start a migration/maintenance window so that no config changes are made in that time. We can't tell them how to do it.